Sources/NIO/Bootstrap.swift

```diff
@@ -66,20 +66,47 @@ public final class ServerBootstrap {
     @usableFromInline
     internal var _childChannelOptions: ChannelOptions.Storage
 
-    /// Create a `ServerBootstrap` for the `EventLoopGroup` `group`.
+    private static func isCompatible(group: EventLoopGroup) -> Bool {
+        return group is SelectableEventLoop || group is MultiThreadedEventLoopGroup
+    }
+
+    /// Create a `ServerBootstrap` on the `EventLoopGroup` `group`.
+    ///
+    /// The `EventLoopGroup` `group` must be compatible, otherwise the program will crash. `ServerBootstrap` is
+    /// compatible only with `MultiThreadedEventLoopGroup` as well as the `EventLoop`s returned by
+    /// `MultiThreadedEventLoopGroup.next`.
     ///
     /// - parameters:
-    ///     - group: The `EventLoopGroup` to use for the `ServerSocketChannel`.
+    ///     - group: The `EventLoopGroup` to use for the `bind` of the `ServerSocketChannel` and to accept new `SocketChannel`s with.
+    ///     - childGroup: The `EventLoopGroup` to run the accepted `SocketChannel`s on.
     public convenience init(group: EventLoopGroup) {
-        self.init(group: group, childGroup: group)
+        guard ServerBootstrap.isCompatible(group: group) else {
+            preconditionFailure("ServerBootstrap is only compatible with MultiThreadedEventLoopGroup and " +
+                                "SelectableEventLoop. You tried constructing one with \(group) which is incompatible.")
+        }
+        self.init(validatingGroup: group, childGroup: group)!
+    }
+
+    public convenience init(group: EventLoopGroup, childGroup: EventLoopGroup) {
+        guard ServerBootstrap.isCompatible(group: group) && ServerBootstrap.isCompatible(group: childGroup) else {
+            preconditionFailure("ServerBootstrap is only compatible with MultiThreadedEventLoopGroup and " +
+                                "SelectableEventLoop. You tried constructing one with \(group) which is incompatible.")
+        }
+        self.init(validatingGroup: group, childGroup: childGroup)!
+
     }
 
-    /// Create a `ServerBootstrap`.
+    /// Create a `ServerBootstrap` on the `EventLoopGroup` `group`, validating that `group` is compatible.
```

Comment:
I think this addition has actually made this more confusing, given that this init takes two groups.